### ISSUE TITLE
osm2pgsql-replication seems not to use env variable PGSERVICE

### ISSUE THREAD

**Issue by gislars** [openstreetmap/osm2pgsql#1800]
## What version of osm2pgsql are you using?

```
osm2pgsql --version
2022-10-26 20:57:12  osm2pgsql version 1.7.0
Build: None
Compiled using the following library versions:
Libosmium 2.18.0
Proj [API 6] 9.1.0
Lua 5.3.6
```

## What operating system and PostgreSQL/PostGIS version are you using?

`Debian GNU/Linux bookworm/sid`

## Tell us something about your system

osm2pgsql is being used in a docker setup. 

## What did you do exactly?

I created a .pg_service.conf file whith following configuration:

```
[osmdb]
host=<db_host>
port=<db_port>
dbname=<db_name>
user=<db_user>
password=<db_password>
```
and used the following commands:
```
export PGSERVICE=osmdb
osm2pgsql-replication init --server https://planet.osm.org/replication/minute/
```

## What did you expect to happen?

The init process should start

## What did happen instead?

The following traceback:

```
osm2pgsql init
Traceback (most recent call last):
  File "/usr/bin/osm2pgsql-replication", line 535, in <module>
    sys.exit(main())
  File "/usr/bin/osm2pgsql-replication", line 527, in main
    conn = connect(args)
  File "/usr/bin/osm2pgsql-replication", line 79, in connect
    if any(part in args.database for part in ['=', '://']):
  File "/usr/bin/osm2pgsql-replication", line 79, in <genexpr>
    if any(part in args.database for part in ['=', '://']):
TypeError: argument of type 'NoneType' is not iterable

```

The init is working if I supply the command line parameters -d -H -U -P

osm2pgsql instead is respecting the env variable PGSERVICE.

## What did you do to try analyzing the problem?
I compared the what happend when I use env variable and command line parameters

**Comment by lonvia** [comment]
This has already been fixed in #1753. Empty database parameters are now allowed. Please update to osm2pgsql 1.7.1 or simply use the replication script from github master.

**Comment by gislars** [comment]
just a follow up: with version 1.7.1 it is working well. I had to change from debian bookworm to sid to test it, it took me a while.
